p384/src/arithmetic/scalar.rs

```diff
@@ -100,6 +100,34 @@ primeorder::impl_mont_field_element!(
 impl Scalar {
     /// Compute [`Scalar`] inversion: `1 / self`.
     pub fn invert(&self) -> CtOption<Self> {
+        #[cfg(all(target_os = "zkvm", target_arch = "riscv32"))]
+        {
+            use crate::elliptic_curve::bigint::Encoding;
+
+            // NOTE: This is not a constant time operation, as inverting zero in the zkvm is not
+            // possible as it will panic in the host.
+            if self.is_zero().into() {
+                return CtOption::new(Scalar::ZERO, Choice::from(0));
+            } else {
+                let input = self.0.to_le_bytes();
+                let input_words = bytemuck::cast::<_, [u32; 12]>(input);
+                let mut output = [0u32; 12];
+                risc0_bigint2::field::modinv_384(
+                    &input_words,
+                    &crate::__risc0::SECP384R1_ORDER,
+                    &mut output,
+                );
+                // PRESTON: doing from_uint_unchecked twice for some reason
```

Comment:
This is a clue that this has something to do with Montgomery form, in that if you have a canonical form field element `a`, with inverse `a^-1`, then in Montgomery form it is `aR` and its inverse is `(a^-1)R` ... and if you performed a canonical form inverse operation on a Montgomery form input `aR`, that would be `(a^-1)(R^-1)`. Then "converting into Montgomery form twice" is just multiplying by `R` twice, and would convert this incorrect inverse to the correct inverse; and, conversely, "converting out of Montgomery form twice" would convert a correct inverse into an incorrect inverse.

So it's some sort of over- or under-conversion into or out of Montgomery form; I will dig further to try to nail down exactly what.